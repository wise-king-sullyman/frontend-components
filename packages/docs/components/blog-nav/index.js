/* eslint-disable react/prop-types */
import { createUseStyles } from 'react-jss';
import Link from 'next/link';
import { MenuContent, MenuItem, MenuList, Title } from '@patternfly/react-core';
import BookOpenIcon from '@patternfly/react-icons/dist/esm/icons/book-open-icon';
import InfoIcon from '@patternfly/react-icons/dist/esm/icons/info-icon'
import classNames from 'classnames';

const useStyles = createUseStyles({
  blogContainer: {
    padding: 8,
  },
  blogLink: {
    color: 'var(--pf-v5-c-menu__list-item--Color)',
  },
  '@media (min-width: 1200px)': {
    blogContainer: {
      position: 'absolute',
      top: 80,
      right: 16,
    },
  },
});

const blogSchema = [
  {
    "title": "Dynamic plugin SDK build",
    "href": "/fed/plugin-sdk",
    Icon: InfoIcon
  },
  {
    "title": "HMR available in development!",
    "href": "/frontend-components-config/hot-module-replacement",
    Icon: InfoIcon
  },

  {
    "title": "React router v6 upgrade",
    "href": "/blog/router-v6",
    Icon: BookOpenIcon
  }]

const BlogNav = () => {
  const classes = useStyles();
  return (
    <div className={classes.blogContainer}>
      <div className="pf-v6-c-menu">
        <MenuContent>
          <MenuList iconSize="large">
            <MenuItem>
              <Title headingLevel="h2">Whats new?</Title>
            </MenuItem>
            {blogSchema.map(({ title, href, Icon }) => (
            <MenuItem
              key={href}
              component={({ props, children }) => (
                <Link href={href}>
                  <button className="pf-v6-c-menu__item">
                    <a {...props} className={classNames(props?.className, classes.blogLink)} href={href}>
                      {children}
                    </a>
                  </button>
                </Link>
              )}
              icon={<Icon />}
            >
              {title}
            </MenuItem>

            ))}
          </MenuList>
        </MenuContent>
      </div>
    </div>
  );
};

export default BlogNav;
