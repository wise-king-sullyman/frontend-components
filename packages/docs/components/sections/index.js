import React from 'react';
import { Bullseye, Card, CardBody, CardTitle, Gallery, GalleryItem, Content, ContentVariants, Title } from '@patternfly/react-core';
import Link from 'next/link';
import { createUseStyles } from 'react-jss';
import classnames from 'classnames';
import truncate from 'lodash/truncate';
import sections from './sections-definition';

const useStyles = createUseStyles({
  description: {
    color: 'black',
    fontSize: "var(--pf-t--global--font--size--sm)",
  },
  cardBody: {
    overflow: 'hidden',
    height: 130,
  },
  link: {
    textDecoration: 'none',
  },
  card: {
    height: 210,
    textDecoration: 'none',
    letterSpacing: 2,
    '& .pf-v6-c-card__title': {
      background: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--BackgroundColor--dark-400 */,
      color: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--Color--light-200 */,
      height: 80,
    },
    '&:hover': {
      '& .pf-v6-c-card__title': {
        background: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--BackgroundColor--dark-200 */,
      },
    },
  },
});

const Sections = () => {
  const classes = useStyles();

  return (
    <Gallery hasGutter className="pf-v6-u-my-2xl">
      {sections.map(({ title, href = '#', description = '' }) => (
        <GalleryItem key={title}>
          <Link href={href}>
            <a className={classes.link}>
              <Card id={title} className={classes.card} isSelectable>
                <CardTitle className={classnames('pf-v6-u-pb-lg')}>
                  <Bullseye>
                    <Title headingLevel="h3">{title}</Title>
                  </Bullseye>
                </CardTitle>
                <CardBody className={classes.cardBody}>
                  <Content className={classnames('pf-v6-u-py-md', classes.description)} component={ContentVariants.p}>
                    {typeof description === 'object' ? description : truncate(description, { length: '135' })}
                  </Content>
                </CardBody>
              </Card>
            </a>
          </Link>
        </GalleryItem>
      ))}
    </Gallery>
  );
};

export default Sections;
