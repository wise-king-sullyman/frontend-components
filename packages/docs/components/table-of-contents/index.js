import { createUseStyles } from 'react-jss';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import { useEffect, useRef, useState } from 'react';
import { Menu, MenuContent, MenuItem, MenuList } from '@patternfly/react-core';

const useStyles = createUseStyles({
  tableOfContent: {
    maxHeight: '100vh',
    overflowY: 'auto',
  },
  list: {
    width: 250,
    boxShadow: 'none !important',
    background: 'none !important',
  },
  listItem: {
    color: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--palette--black-700 */,
    '&:hover': {
      color: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--palette--black-900 */,
    },
  },
  activeLink: {
    color: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--palette--black-900 */,
    background: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--palette--white */,
    position: 'relative',
    '&::before': {
      content: '""',
      position: 'absolute',
      left: -8,
      top: 0,
      bottom: 0,
      borderLeft: "var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--palette--blue-400 */,
    },
  },
});

const ContentLink = ({ title, setActive, isActive, targetId }) => {
  const classes = useStyles();
  return (
    <MenuItem
      onClick={() => {
        setActive(targetId);
      }}
      title={title}
      className={classnames(classes.listItem, {
        [classes.activeLink]: isActive,
      })}
      to={`#${targetId}`}
    >
      {title}
    </MenuItem>
  );
};

ContentLink.propTypes = {
  title: PropTypes.node.isRequired,
  targetId: PropTypes.string.isRequired,
  isActive: PropTypes.bool,
  setActive: PropTypes.func.isRequired,
};

const TableOfContents = () => {
  const tocRef = useRef(null);
  const classes = useStyles();
  const { pathname } = useRouter();
  const [links, setLinks] = useState([]);
  const [activeItem, setActive] = useState();
  let isMounted = true;

  const scrollListener = (setActive) => {
    const contentElement = document.getElementById('docs-content');
    const anchors = Array.from(contentElement.getElementsByClassName('docs-content-link'));
    const min = -20;
    const max = 20;
    const elem = anchors.find((elem) => {
      const { top } = elem.getBoundingClientRect();
      return top > min && top < max;
    });
    if (isMounted && elem) {
      setActive(elem.firstElementChild.id);
    }
  };

  useEffect(() => {
    if (tocRef.current) {
      tocRef.current.style.position = 'sticky';
      tocRef.current.style.right = '20px';
      tocRef.current.style.top = '0px';
    }
  }, []);

  useEffect(() => {
    // eslint-disable-next-line react-hooks/exhaustive-deps
    isMounted = true;
    document.querySelector('.pf-v6-c-page__main').addEventListener('scroll', () => scrollListener(setActive));
    scrollListener(setActive);

    return () => {
      isMounted = false;
      document.removeEventListener('scroll', scrollListener);
    };
  }, []);

  useEffect(() => {
    const contentElement = document.getElementById('docs-content');
    const anchors = Array.from(contentElement.getElementsByClassName('docs-content-link'));
    const links = anchors.map((elem) => ({
      component: elem.tagName,
      title: elem.textContent,
      targetId: elem.firstElementChild.id,
      level: Number(elem.tagName.replace('H', '')) - 1,
    }));
    setLinks(links);
  }, [pathname]);

  return links.length > 0 ? (
    <div ref={tocRef}>
      <Menu className={classes.list}>
        <MenuContent>
          <MenuList className={classes.tableOfContent}>
            {/* eslint-disable react/prop-types */}
            {links.map((props) => (
              <ContentLink setActive={setActive} key={props.targetId} isActive={props.targetId === activeItem} {...props} />
            ))}
            {/* eslint-enable react/prop-types */}
          </MenuList>
        </MenuContent>
      </Menu>
    </div>
  ) : null;
};

export default TableOfContents;
