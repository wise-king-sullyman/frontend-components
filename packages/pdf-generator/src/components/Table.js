/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from '@react-pdf/renderer';
import color_gray_10 from '@patternfly/react-tokens/dist/js/color_gray_10';
import color_gray_50 from '@patternfly/react-tokens/dist/js/color_gray_50';
import { styleProps } from '../utils/propTypes';
import styles from '../utils/styles';

const appliedStyles = styles();

const Table = ({ rows, withHeader, headerStyles, rowsStyle, style, ...props }) => {
  const [header, ...cells] = rows;
  return (
    <View
      {...props}
      style={{
        ...appliedStyles.flexColum,
        ...style,
      }}
    >
      {withHeader && (
        <View
          style={{
            ...appliedStyles.flexRow,
            ...appliedStyles.compactCellPadding,
            ...headerStyles,
          }}
        >
          {header.map((cell, key) =>
            typeof cell === 'string' || cell instanceof String ? (
              <Text
                key={key}
                style={{
                  ...appliedStyles.secondTitle,
                  ...appliedStyles.compactCellPadding,
                  flex: 1,
                }}
              >
                {cell}
              </Text>
            ) : (
              cell
            )
          )}
        </View>
      )}
      <View
        style={{
          borderTopStyle: 'solid',
          borderTopColor: color_gray_50.value,
          borderTopWidth: 1,
          borderBottomStyle: 'solid',
          borderBottomColor: color_gray_50.value,
          borderBottomWidth: 1,
        }}
      >
        {(!withHeader ? [header, ...cells] : cells).map((row, key) => (
          <View
            key={key}
            style={{
              ...appliedStyles.flexRow,
              ...appliedStyles.compactCellPadding,
              ...(key % 2 && { backgroundColor: color_gray_10.value }),
              ...rowsStyle,
            }}
          >
            {row.map((cell, cellKey) =>
              typeof cell === 'string' || cell instanceof String ? (
                <Text
                  key={cellKey}
                  style={{
                    ...appliedStyles.text,
                    ...appliedStyles.compactCellPadding,
                    flex: 1,
                  }}
                >
                  {cell}
                </Text>
              ) : (
                cell
              )
            )}
          </View>
        ))}
      </View>
    </View>
  );
};

Table.propTypes = {
  style: styleProps,
  rowsStyle: styleProps,
  withHeader: PropTypes.bool,
  rows: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.node)),
  headerStyles: styleProps,
};

Table.defaultProps = {
  style: {},
  headerStyles: {},
  rowsStyle: {},
};

export default Table;
