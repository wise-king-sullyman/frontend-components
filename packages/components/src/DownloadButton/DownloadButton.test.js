import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import DownloadButton from './DownloadButton';
import { MenuItem } from '@patternfly/react-core/dist/dynamic/components/Menu';
import { act } from 'react-dom/test-utils';

const extraItems = [<MenuItem key="extra-1" component="button"></MenuItem>];

describe('DownloadButton component', () => {
  describe('should render', () => {
    it('CSV and JSON by default', () => {
      const { container } = render(<DownloadButton />);
      expect(container).toMatchSnapshot();
    });

    it('custom items', () => {
      const { container } = render(<DownloadButton extraItems={extraItems} />);
      expect(container).toMatchSnapshot();
    });

    it('disabled', () => {
      const { container } = render(<DownloadButton extraItems={extraItems} isDisabled />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('API', () => {
    it('clicking should open dropdown', async () => {
      const { container } = render(<DownloadButton extraItems={extraItems} />);
      await act(async () => {
        userEvent.click(screen.getByRole('button', { name: 'Export' }));
      });
      expect(container).toMatchSnapshot();
    });

    it('onSelect should be called with CSV', async () => {
      const onSelect = jest.fn();
      render(<DownloadButton extraItems={extraItems} onSelect={onSelect} />);
      await act(async () => {
        userEvent.click(screen.getByRole('button', { name: 'Export' }));
      });

      act(() => {
        userEvent.click(screen.getByRole('menuitem', { name: 'Export to CSV' }));
      });
      expect(onSelect.mock.calls.length).toBe(1);
      expect(onSelect.mock.calls[0][1]).toBe('csv');
    });

    it('onSelect should be called with JSON', async () => {
      const onSelect = jest.fn();
      render(<DownloadButton extraItems={extraItems} onSelect={onSelect} />);
      await act(async () => {
        userEvent.click(screen.getByRole('button', { name: 'Export' }));
      });

      act(() => {
        userEvent.click(screen.getByRole('menuitem', { name: 'Export to JSON' }));
      });
      expect(onSelect.mock.calls.length).toBe(1);
      expect(onSelect.mock.calls[0][1]).toBe('json');
    });

    it("shouldn't call onSelect", async () => {
      const onSelect = jest.fn();
      render(<DownloadButton extraItems={extraItems} />);
      await act(async () => {
        userEvent.click(screen.getByRole('button', { name: 'Export' }));
      });

      act(() => {
        userEvent.click(screen.getByRole('menuitem', { name: 'Export to JSON' }));
      });
      expect(onSelect.mock.calls.length).toBe(0);
    });
  });
});
