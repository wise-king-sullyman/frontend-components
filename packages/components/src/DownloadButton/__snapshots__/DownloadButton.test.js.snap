// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton component API clicking should open dropdown 1`] = `
<div>
  <div
    style="display: contents;"
  >
    <button
      aria-expanded="true"
      aria-label="Export"
      class="pf-v6-c-menu-toggle pf-m-expanded pf-m-plain"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plain-4"
      data-ouia-component-type="PF6/MenuToggle"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-menu-toggle__icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v6-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M975.8,636.9 L870.9,741.8 L457.9,328.6 C452.1,322.8 445.4,319.9 437.9,319.9 C430.4,319.9 423.7,322.8 417.9,328.6 L328.8,417.7 C323,423.5 320.1,430.2 320.1,437.7 C320.1,445.2 323,451.9 328.8,457.7 L742,870.7 L636.9,975.8 C610.5,1002.2 619.4,1024 656.8,1024 L956,1024 C1014.5,1024 1024,1013.7 1024,955.9 L1024,656.7 C1023.9,619.4 1002.2,610.5 975.8,636.9 Z M128,128 L896,128 L896,361.7 C896.007942,370.182681 899.389907,378.313788 905.4,384.3 L996.7,475.6 C1006.8,485.7 1024,478.5 1024,464.3 L1024,22.7 C1024,16.1 1021.9,10.7 1017.6,6.4 C1013.3,2.1 1007.9,0 1001.3,0 L22.7,0 C16.1,0 10.7,2.1 6.4,6.4 C2.1,10.7 0,16.1 0,22.7 L0,1001.3 C0,1007.9 2.1,1013.3 6.4,1017.6 C10.7,1021.9 16.1,1024 22.7,1024 L463.4,1024 C469.862884,1023.98894 475.684489,1020.0908 478.156232,1014.11925 C480.627976,1008.14769 479.264428,1001.27548 474.7,996.7 L383.4,905.4 C377.413788,899.389907 369.282681,896.007942 360.8,896 L128,896 L128,128 Z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`DownloadButton component should render CSV and JSON by default 1`] = `
<div>
  <div
    style="display: contents;"
  >
    <button
      aria-expanded="false"
      aria-label="Export"
      class="pf-v6-c-menu-toggle pf-m-plain"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plain-1"
      data-ouia-component-type="PF6/MenuToggle"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-menu-toggle__icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v6-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M975.8,636.9 L870.9,741.8 L457.9,328.6 C452.1,322.8 445.4,319.9 437.9,319.9 C430.4,319.9 423.7,322.8 417.9,328.6 L328.8,417.7 C323,423.5 320.1,430.2 320.1,437.7 C320.1,445.2 323,451.9 328.8,457.7 L742,870.7 L636.9,975.8 C610.5,1002.2 619.4,1024 656.8,1024 L956,1024 C1014.5,1024 1024,1013.7 1024,955.9 L1024,656.7 C1023.9,619.4 1002.2,610.5 975.8,636.9 Z M128,128 L896,128 L896,361.7 C896.007942,370.182681 899.389907,378.313788 905.4,384.3 L996.7,475.6 C1006.8,485.7 1024,478.5 1024,464.3 L1024,22.7 C1024,16.1 1021.9,10.7 1017.6,6.4 C1013.3,2.1 1007.9,0 1001.3,0 L22.7,0 C16.1,0 10.7,2.1 6.4,6.4 C2.1,10.7 0,16.1 0,22.7 L0,1001.3 C0,1007.9 2.1,1013.3 6.4,1017.6 C10.7,1021.9 16.1,1024 22.7,1024 L463.4,1024 C469.862884,1023.98894 475.684489,1020.0908 478.156232,1014.11925 C480.627976,1008.14769 479.264428,1001.27548 474.7,996.7 L383.4,905.4 C377.413788,899.389907 369.282681,896.007942 360.8,896 L128,896 L128,128 Z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`DownloadButton component should render custom items 1`] = `
<div>
  <div
    style="display: contents;"
  >
    <button
      aria-expanded="false"
      aria-label="Export"
      class="pf-v6-c-menu-toggle pf-m-plain"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plain-2"
      data-ouia-component-type="PF6/MenuToggle"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-menu-toggle__icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v6-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M975.8,636.9 L870.9,741.8 L457.9,328.6 C452.1,322.8 445.4,319.9 437.9,319.9 C430.4,319.9 423.7,322.8 417.9,328.6 L328.8,417.7 C323,423.5 320.1,430.2 320.1,437.7 C320.1,445.2 323,451.9 328.8,457.7 L742,870.7 L636.9,975.8 C610.5,1002.2 619.4,1024 656.8,1024 L956,1024 C1014.5,1024 1024,1013.7 1024,955.9 L1024,656.7 C1023.9,619.4 1002.2,610.5 975.8,636.9 Z M128,128 L896,128 L896,361.7 C896.007942,370.182681 899.389907,378.313788 905.4,384.3 L996.7,475.6 C1006.8,485.7 1024,478.5 1024,464.3 L1024,22.7 C1024,16.1 1021.9,10.7 1017.6,6.4 C1013.3,2.1 1007.9,0 1001.3,0 L22.7,0 C16.1,0 10.7,2.1 6.4,6.4 C2.1,10.7 0,16.1 0,22.7 L0,1001.3 C0,1007.9 2.1,1013.3 6.4,1017.6 C10.7,1021.9 16.1,1024 22.7,1024 L463.4,1024 C469.862884,1023.98894 475.684489,1020.0908 478.156232,1014.11925 C480.627976,1008.14769 479.264428,1001.27548 474.7,996.7 L383.4,905.4 C377.413788,899.389907 369.282681,896.007942 360.8,896 L128,896 L128,128 Z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`DownloadButton component should render disabled 1`] = `
<div>
  <div
    style="display: contents;"
  >
    <button
      aria-expanded="false"
      aria-label="Export"
      class="pf-v6-c-menu-toggle pf-m-plain"
      data-ouia-component-id="OUIA-Generated-MenuToggle-plain-3"
      data-ouia-component-type="PF6/MenuToggle"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-menu-toggle__icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v6-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M975.8,636.9 L870.9,741.8 L457.9,328.6 C452.1,322.8 445.4,319.9 437.9,319.9 C430.4,319.9 423.7,322.8 417.9,328.6 L328.8,417.7 C323,423.5 320.1,430.2 320.1,437.7 C320.1,445.2 323,451.9 328.8,457.7 L742,870.7 L636.9,975.8 C610.5,1002.2 619.4,1024 656.8,1024 L956,1024 C1014.5,1024 1024,1013.7 1024,955.9 L1024,656.7 C1023.9,619.4 1002.2,610.5 975.8,636.9 Z M128,128 L896,128 L896,361.7 C896.007942,370.182681 899.389907,378.313788 905.4,384.3 L996.7,475.6 C1006.8,485.7 1024,478.5 1024,464.3 L1024,22.7 C1024,16.1 1021.9,10.7 1017.6,6.4 C1013.3,2.1 1007.9,0 1001.3,0 L22.7,0 C16.1,0 10.7,2.1 6.4,6.4 C2.1,10.7 0,16.1 0,22.7 L0,1001.3 C0,1007.9 2.1,1013.3 6.4,1017.6 C10.7,1021.9 16.1,1024 22.7,1024 L463.4,1024 C469.862884,1023.98894 475.684489,1020.0908 478.156232,1014.11925 C480.627976,1008.14769 479.264428,1001.27548 474.7,996.7 L383.4,905.4 C377.413788,899.389907 369.282681,896.007942 360.8,896 L128,896 L128,128 Z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
